app/src/main/java/com/infomaniak/mail/data/models/Attachment.kt

```diff
@@ -72,12 +72,21 @@ class Attachment : EmbeddedRealmObject {
 
     inline val safeMimeType get() = if (mimeType == MIMETYPE_UNKNOWN) name.guessMimeType() else mimeType
 
-    fun initLocalValues(name: String, size: Long, mimeType: String, uri: String): Attachment {
-        this.name = name
-        this.size = size
-        this.mimeType = mimeType
+    fun initLocalValues(
+        uri: String?,
+        uuid: String? = null,
+        name: String? = null,
+        size: Long? = null,
+        mimeType: String? = null,
+    ): Attachment {
+
         this.uploadLocalUri = uri
 
+        uuid?.let { localUuid = it }
+        name?.let { this.name }
+        size?.let { this.size = it }
+        mimeType?.let { this.mimeType = it }
+
```

Comment:
I'm not a fan of having all these nullable arguments here. I think that when we actually use it to instantiate an object it should keep all of its non-null values and when we want to use it to do something else, like backup some values we could at least create another method explicitly used for this very and only purpose